unison-runtime/src/Unison/Runtime/Foreign.hs

```diff
@@ -358,6 +359,15 @@ instance BuiltinForeign CharPattern where
   foreignName = Tagged "CharPattern"
   foreignRef = Tagged Ty.charClassRef
 
+-- Note: this doesn't do any recursive conversion of keys/values,
+-- so any use of it needs to be an exact match for the map that was
+-- originally placed in the box. The current intention is for use
+-- at `Map Val Val`, but `Val` is in a module further along the
+-- dependency graph.
```

Comment:
Would it maybe be better to just define the `Map Val Val` instance in the module with `Val` then?

Reply:
Hmm, maybe.